### PR TITLE
FEATURE: Add onebox support for chat threads

### PR DESCRIPTION
**Pull request by jancernik**
<div align="center">

### Thread Messages

</div>

### CODE REVIEW

File: plugins/chat/config/locales/server.en.yml [hunk 1]
```diff
@@ -176,6 +176,8 @@ en:
       inline_to_message: "Message #%{message_id} by %{username} – #%{chat_channel}"
       inline_to_channel: "Chat #%{chat_channel}"
       inline_to_topic_channel: "Chat for Topic %{topic_title}"
+      default_thread_title: "Thread"
+      thread_title_connector: "in"
```
Comment (jancernik) [inline plugins/chat/config/locales/server.en.yml:180]: This doesn't seem right to me.
```
thread_title_connector: I18n.t("chat.onebox.thread_title_connector"),
```


```
<span class="thread-title-connector">{{{thread_title_connector}}}</span>
```
Is there a better way to do this?

**Comment by jancernik** [comment]
@jjaffeux @chapoi
This PR also adds images to regular messages oneboxes, in additon to the new threads oneboxes, but I'm not too sure about the aproach I went for. Note that it would only include one image, even if the message contains more.

File: plugins/chat/config/locales/server.en.yml (hunk 1 above)
Comment (jjaffeux) [reply plugins/chat/config/locales/server.en.yml:180]: We don't use default thread title anymore IMO, what do you say @chapoi ?

Comment (chapoi) [reply plugins/chat/config/locales/server.en.yml:180]: Yes, just leave it blank. I have an upcoming design idea for it anyway.

I do think the images are looking very big, might be the perspective of the screenshots, but might need to be scaled down a little. Will be more clear when it's in live, so don't let that block.

File: plugins/chat/lib/onebox/templates/discourse_chat_thread.mustache [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<aside class="onebox chat-onebox">
```
Comment (chapoi) [inline plugins/chat/lib/onebox/templates/discourse_chat_thread.mustache:1]: Bit strange choice here for aside.

File: plugins/chat/config/locales/server.en.yml (hunk 1 above)
Comment (jjaffeux) [reply plugins/chat/config/locales/server.en.yml:180]: Yes agree on the size of images @jancernik. Maybe we could have some gallery like solution?

Screenshot 2023-09-26 at 14 06 06

Something like this?

File: plugins/chat/lib/onebox/templates/discourse_chat_thread.mustache (hunk 2 above)
Comment (jancernik) [reply plugins/chat/lib/onebox/templates/discourse_chat_thread.mustache:1]: It's part of the default onebox styling :man_shrugging: 
https://github.com/discourse/discourse/blob/7019dbcfe123626c279e3f3f5733683eaec4008f/app/assets/stylesheets/common/base/onebox.scss#L94-L315

File: plugins/chat/lib/chat/onebox_handler.rb [hunk 3]
```diff
@@ -66,6 +88,17 @@ def self.render_channel_onebox(args, chat_channel)
       Mustache.render(Chat.channel_onebox_template, args)
     end
 
+    def self.get_image_uploads(target)
+      if target.is_a?(Message)
+        message = target
+      elsif target.is_a?(Thread)
+        message = Chat::Message.find_by(id: target.original_message_id)
```
Comment (jjaffeux) [inline plugins/chat/lib/chat/onebox_handler.rb:95]: Maybe includes(:uploads) here so we don't do another query after

File: plugins/chat/spec/lib/chat/onebox_handler_spec.rb [hunk 4]
```diff
@@ -133,4 +133,96 @@
...
+          Chat::OneboxHandler.handle(
+            "#{private_chat_url}/t/#{private_thread.id}",
+            { channel_id: private_channel.id, thread_id: public_thread.id },
+          )
+
+        expect(onebox_html).to be_nil
+      end
+    end
+
+    context "when thread does not exists" do
```
Comment (jjaffeux) [inline plugins/chat/spec/lib/chat/onebox_handler_spec.rb:194]:
```suggestion
    context "when thread does not exist" do
```

File: plugins/chat/lib/chat/onebox_handler.rb (hunk 3 above)
Comment (jjaffeux) [inline plugins/chat/lib/chat/onebox_handler.rb:95]:
```suggestion
        message = Chat::Message.includes(:uploads).find_by(id: target.original_message_id)
```

**Review by jjaffeux — APPROVED** [review]

**Pull request merged**